src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java

```diff
@@ -377,6 +394,11 @@ public String getCredentialsId() {
         return credentialsId;
     }
 
+    @CheckForNull
+    public String getCredentialsIdRaw() {
```

Comment:
Like I commented on the form, I think it would be best if we could just reuse `credentialsId` and delete this field.

Reply:
+1

Reply:
As discussed elsewhere with Jose, the problem is that if the form has two inputs with the same name (one for each radio button), Jenkins thinks that that field should be treated as a list, and so it submits an array of values, one for everywhere it appears, and that breaks the form binding. Maybe possible to work around this with JS, but needs to be investigated.

Reply:
How about we pull out credentials from the radio button group, its common element anyways? Something like:

<img width="1048" alt="image" src="https://user-images.githubusercontent.com/38139/60454493-cc129a80-9be8-11e9-9893-e710a87302d9.png">


```patch
<j:jelly xmlns:j="jelly:core" xmlns:c="/lib/credentials"
          xmlns:f="/lib/form"
          xmlns:scm="/jenkins/scm/api/form">
+      <f:entry title="${%Credentials}" field="credentialsId">
+        <c:select checkMethod="post"/>
+      </f:entry>
+
   <f:radioBlock name="url" value="Github Repository" checked="${instance.isConfiguredByRepoScan()}" title="Github Repository Scan" inline="true">
     <f:nested>
       <j:choose>
@@ -16,9 +20,6 @@
           </f:invisibleEntry>
         </j:otherwise>
       </j:choose>
-      <f:entry title="${%Credentials}" field="credentialsId">
-        <c:select checkMethod="post"/>
-      </f:entry>
       <f:entry title ="${%Owner}" field="repoOwner">
         <f:textbox/>
       </f:entry>
@@ -30,9 +31,6 @@
   </f:radioBlock>
   <f:radioBlock name="url" value="Github Repository" checked="${!instance.isConfiguredByRepoScan()}" title="Github Repository HTTPs URL" inline="true">
     <f:nested>
-      <f:entry title="${%Credentials}" field="credentialsIdRaw">
-        <c:select checkMethod="post"/>
-      </f:entry>
       <f:entry title="HTTPS URL" field="rawUrl">
         <f:textbox/>
       </f:entry>
```

Reply:
@joseblas @dwnusbaum If pulling out credentials outside of radio button groups makes sense and works in all cases, we should run it by Joe/UX as well.

Reply:
@Vivek lgtm, I'll ping him to see whether he likes it

Reply:
As discussed with @joseblas and @josephbrueggen, sharing the credentials across the two configuration modes would require, in the new UX, to display the credentials selector before displaying the input textbox to enter the github URL. It may be slightly counter intuitive but this sounds good to me as it allows to verify the permissions on the github repository just after the URL has been provided.